src/apps/statements/repo/eventsRepo/emitNewStatements/sqs.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  GetQueueUrlCommand,
+  SQSClient,
+  SendMessageBatchCommand,
+  SendMessageBatchRequestEntry,
+} from '@aws-sdk/client-sqs';
+import { v4 } from 'uuid';
+import { getPrefixWithProcessingPriority } from '../utils/getPrefixWithProcessingPriority';
+import { StatementProcessingPriority } from '../../../enums/statementProcessingPriority.enum';
+import FacadeConfig from '../utils/sqsEvents/FacadeConfig';
+import { STATEMENT_QUEUE } from '../utils/constants';
+import Signature from './Signature';
+
+const MAX_BATCH_SIZE = 10;
+
+let queueUrl: string | undefined;
```

Comment:
Would be nice to avoid this `let`, do we need to allow for it to be `undefined`?
Maybe we should wait for line 67 to resolve and pass `queueUrl` as a parameter to `publishMessages`? Or is there a reason for not doing that?

Reply:
I did this just to avoid unnecessary AWS requests and storing queueUrl in [memory](https://github.com/LearningLocker/xapi-service/pull/890/files#diff-56443e31728bc62625f68d5b91d8d3fb0ac604bb552260679dd69fa91ef31f7dR42-R44)

Reply:
Anyway, feel free to post your solution, please

Reply:
Yeah ok 👍 
The alternative would be:

```
const publishMessages = async (sqsClient: SQSClient, statementProperties: string[], queueUrl: string | undefined) => { ... };

const getQueueUrl = async (
  sqsClient: SQSClient,
  ...
) => {
  const prefixWithPriority = getPrefixWithProcessingPriority(
    prefix,
    priority,
    isQueuePriorityEnabled,
  );
  ...
};

export default (config: FacadeConfig): Signature => {
  return async ({ statementProperties, priority }) => {
    const sqsClient = await config.client();

    const queueUrl = await getQueueUrl(sqsClient, config.prefix, priority, config.isQueuePriorityEnabled);
    await publishMessages(sqsClient, statementProperties, queueUrl);
  };
};

```

But I don't think this is better - as you say, you wouldn't be able to pull `queueUrl` from memory (also you still need the undefined in definition).
I'll approve this, thanks for explaining 🙂